avm/res/compute/virtual-machine/main.bicep

```diff
@@ -622,24 +595,36 @@ resource vm_configurationProfileAssignment 'Microsoft.Automanage/configurationPr
   }
 
 resource vm_autoShutdownConfiguration 'Microsoft.DevTestLab/schedules@2018-09-15' =
-  if (autoShutdownStatus == 'Enabled') {
+  if (!empty(autoShutdownConfig)) {
     name: 'shutdown-computevm-${vm.name}'
     location: location
     properties: {
-      status: autoShutdownStatus
+      status: contains(autoShutdownConfig, 'status') ? autoShutdownConfig.status : 'Disabled'
       targetResourceId: vm.id
       taskType: 'ComputeVmShutdownTask'
       dailyRecurrence: {
-        time: autoShutdownTime
-      }
-      timeZoneId: autoShutdownTimeZone
-      notificationSettings: {
-        status: autoShutdownNotificationStatus
-        emailRecipient: autoShutdownNotificationEmail
-        notificationLocale: autoShutdownNotificationLocale
-        timeInMinutes: autoShutdownNotificationTimeInMinutes
-        webhookUrl: autoShutdownWebhookUrl
+        time: contains(autoShutdownConfig, 'time') ? autoShutdownConfig.time : '19:00'
       }
+      timeZoneId: contains(autoShutdownConfig, 'timeZone') ? autoShutdownConfig.timeZone : 'UTC'
+      notificationSettings: contains(autoShutdownConfig, 'notificationStatus')
+        ? {
+            status: contains(autoShutdownConfig, 'notificationStatus')
+              ? autoShutdownConfig.notificationStatus
+              : 'Disabled'
+            emailRecipient: contains(autoShutdownConfig, 'notificationEmail')
+              ? autoShutdownConfig.notificationEmail
+              : ''
+            notificationLocale: contains(autoShutdownConfig, 'notificationLocale')
+              ? autoShutdownConfig.notificationLocale
+              : 'en'
+            timeInMinutes: contains(autoShutdownConfig, 'notificationWebhookUrl')
+              ? autoShutdownConfig.notificationWebhookUrl
+              : ''
+            webhookUrl: contains(autoShutdownConfig, 'notificationTimeInMinutes')
+              ? autoShutdownConfig.notificationTimeInMinutes
+              : 30
```

Comment:
Are we sure about all these default values?

Reply:
@eriqua  That is what you get in the portal experience

Reply:
Fair. When implementing user defined types we'll be able to properly document those defaults